server/athenian/api/internal/features/entries.py

```diff
@@ -121,15 +121,41 @@
 
 
 def _postprocess_cached_facts(
-    result: Tuple[Dict[str, List[PullRequestFacts]], bool],
+    result: tuple[dict[str, list[PullRequestFacts]], bool],
     with_jira_map: bool,
     **_,
-) -> Tuple[Dict[str, List[PullRequestFacts]], bool]:
+) -> tuple[dict[str, list[PullRequestFacts]], bool]:
     if with_jira_map and not result[1]:
         raise CancelCache()
     return result
 
 
+class _PRMetricsLineGHCache:
+    @staticmethod
+    def postprocess(result: np.ndarray, *, metrics: Sequence[str], **kwargs: Any) -> np.ndarray:
+        # cache value has been stored with sorted(metrics), adapt before returning to caller
+        if (sorted_metrics := sorted(metrics)) != metrics:
+            sort_indexes = [metrics.index(m) for m in sorted_metrics]
+            return _sorted_result_metrics(result, sort_indexes)
+
+        return result
+
+    @staticmethod
+    def preprocess(result: np.ndarray, *, metrics: Sequence[str], **kwargs: Any) -> np.ndarray:
+        # cache value must be stored with sorted(metrics), adapt if needed
+        if (sorted_metrics := sorted(metrics)) != metrics:
+            sort_indexes = np.searchsorted(sorted_metrics, metrics)
+            return _sorted_result_metrics(result, sort_indexes)
+        return result
+
+
+def _sorted_result_metrics(result: np.ndarray, metrics_sort_indexes: Iterable[int]) -> np.ndarray:
+    copy = result.reshape(-1).copy()
```

Comment:
Can you please use `ravel()` - it is more idiomatic than remembering the magic argument values.